pkg/instancegroups/rollingupdate.go

```diff
@@ -69,6 +69,31 @@ type RollingUpdateCluster struct {
 	ValidateSuccessDuration time.Duration
 }
 
+// AdjustNeedUpdate adjusts the set of instances that need updating, using factors outside those known by the cloud implementation
+func (c *RollingUpdateCluster) AdjustNeedUpdate(groups map[string]*cloudinstances.CloudInstanceGroup, cluster *api.Cluster, instanceGroups *api.InstanceGroupList) error {
+	for _, group := range groups {
+		if group.Ready != nil {
```

Comment:
FYI for reads golang treats a nil slice as empty, so I don't think you actually need this check.  Arguably it does make things clearer though!